### PR TITLE
End to end test, HomeScreen

### PR DESCRIPTION
**Pull request by UnavailableCode**
### END2END : HomeScreen 
The end to end test now checks that components are properly displayed in the home screen and clicks on the profile picture and on the search bar.

### CODE REVIEW

File: app/src/androidTest/java/com/example/triptracker/e2eTest.kt [hunk 1]
```diff
@@ -121,19 +261,5 @@ class E2ETest {
     // Go back and assert we are in the profile screen
     composeTestRule.onNodeWithTag("GoBackButton").performClick()
     composeTestRule.onNodeWithTag("ProfileOverview").assertIsDisplayed()
-
```
Comment (SpaceMercury) [inline app/src/androidTest/java/com/example/triptracker/e2eTest.kt:124]: Why are these lines removed?

File: app/src/androidTest/java/com/example/triptracker/e2eTest.kt [hunk 2]
```diff
@@ -1,58 +1,141 @@
...
+        {
+          // Invoke the callback with mock data
+          val callback = arg<(List<UserProfile>) -> Unit>(0)
+          callback(mockUsers)
+        }
+  }
 
-  /*
-     Test to check the flow of the profile screen navigating through different screens
-  */
```
Comment (SpaceMercury) [inline app/src/androidTest/java/com/example/triptracker/e2eTest.kt:43]: Why did you remove the comment? Should we add a different comment instead?

**Review by SpaceMercury — CHANGES_REQUESTED** [review]
CI is not passing take a look! Apart from that good testing!

**Review by SpaceMercury — APPROVED** [review]
CI is not passing take a look! Apart from that good testing!

File: app/src/androidTest/java/com/example/triptracker/e2eTest.kt (hunk 2 above)
Comment (UnavailableCode) [reply app/src/androidTest/java/com/example/triptracker/e2eTest.kt:43]: Because its not about the profile screen anymore but a global end to end !

File: app/src/androidTest/java/com/example/triptracker/e2eTest.kt (hunk 1 above)
Comment (UnavailableCode) [reply app/src/androidTest/java/com/example/triptracker/e2eTest.kt:124]: This was an accident on my side, let's fix that !
